### PR TITLE
Add doc for statistics

### PR DESCRIPTION
**Pull request by losipiuk**
Introduces documentation for statistics mechanism added by https://github.com/prestodb/presto/pull/7212

### CODE REVIEW

File: presto-docs/src/main/sphinx/optimizer/statistics.rst [hunk 1]
```diff
@@ -0,0 +1,144 @@
+================
+Table Statistics
+================
+
+Presto supports statistics based optimizations for queries. For a query to take advantage of these optimizations,
+Presto must have statistical information for the tables in that query.
+
+Table statistics are provided to the query planner by connectors.
+Currently the only connector that supports statistics is the Hive connector.
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/optimizer/statistics.rst]: Add a section link to Hive connector.

File: presto-docs/src/main/sphinx/optimizer/statistics.rst [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+Table Layouts
+-------------
+
+Statistics are exposed to the query planner by table layout. A table layout represents a subset of a table's data
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/optimizer/statistics.rst]: s/by table/by a table layout

File: presto-docs/src/main/sphinx/optimizer/statistics.rst [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+* Partitioned tables have a family of table layouts. Each set of partitions to be scanned represents one table layout.
+  Presto will try to pick a table layout consisting of the smallest number of partitions based on filtering predicates
+  from the query.
+
+Available Statistics
+--------------------
+
+Currently, the following statistics are available in Presto:
+
+ * For the whole table:
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/optimizer/statistics.rst]: Just say "For a table"

File: presto-docs/src/main/sphinx/optimizer/statistics.rst [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+   * **row count**: the total number of rows for the table layout
+
+ * For each column:
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/optimizer/statistics.rst]: For each column in a table

File: presto-docs/src/main/sphinx/optimizer/statistics.rst [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+ * ``SHOW STATS FOR <table_name>`` will show statistics for the table layout representing all data in the table
+ * ``SHOW STATS FOR (SELECT <column_list|*> FROM <table_name> WHERE <filtering_condition>)``
+   will show statistics for the table layout of table ``t`` representing a subset of data after applying the given filtering
+   condition. Both the column list and the filtering condition used in the ``WHERE`` clause can reference table columns.
+
+In both cases, the ``SHOW STATS`` command outputs two types of rows.
+For each column in the table there is a row with ``column_name`` equal to the name of that column.
+These rows expose column-related statistics for a table (data size, nulls count, distinct values count).
+Additionally there is one row with NULL as the ``column_name``. This row contains table-layout wide statistics - for now just row count.
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/optimizer/statistics.rst]: just the row count

File: presto-docs/src/main/sphinx/optimizer/statistics.rst [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    -------------+-----------+-----------------------+-------------+-----------
+     comment     | NULL      |                  21.0 |        21.0 | NULL
+     nationkey   | NULL      |                   9.0 |         9.0 | NULL
+     name        | NULL      |                  14.0 |        14.0 | NULL
+     regionkey   | NULL      |                   3.0 |         3.0 | NULL
+     NULL        | NULL      | NULL                  | NULL        |      15.0
+     (5 rows)
+
+If provided ``SELECT`` will filter out all of the partitions (all table layouts),
+than the ``SHOW STATS`` will return no statistic which will be represented as in example below.
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/optimizer/statistics.rst]: s/than/then

File: presto-docs/src/main/sphinx/optimizer/statistics.rst [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+    -------------+-----------+-----------------------+-------------+-----------
+     comment     | NULL      |                  21.0 |        21.0 | NULL
+     nationkey   | NULL      |                   9.0 |         9.0 | NULL
+     name        | NULL      |                  14.0 |        14.0 | NULL
+     regionkey   | NULL      |                   3.0 |         3.0 | NULL
+     NULL        | NULL      | NULL                  | NULL        |      15.0
+     (5 rows)
+
+
+Updating Statistics for Hive Tables
```
Comment (akshatnair) [inline presto-docs/src/main/sphinx/optimizer/statistics.rst]: s/for/For

File: presto-docs/src/main/sphinx/optimizer/cost-in-explain.rst [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+                LAYOUT: hive
+                nationkey := HiveColumnHandle{clientId=hive, name=nationkey, hiveType=bigint, hiveColumnIndex=0, columnType=REGULAR}
+                comment := HiveColumnHandle{clientId=hive, name=comment, hiveType=varchar(152), hiveColumnIndex=3, columnType=REGULAR}
+
+Generally there is only one cost printed for each plan node.
+However, when a ``Scan`` operator is combined with a ``Filter`` and/or ``Project`` operator, then multiple cost structures will be printed,
+each corresponding to an individual logical part of the combined meta-operator.
+For example, for a ``ScanFilterProject`` operator three cost structures will be printed.
+
+ * the first will correspond to ``Scan`` part of operator
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/optimizer/cost-in-explain.rst]: The example above visualises that.

File: presto-docs/src/main/sphinx/optimizer/cost-in-explain.rst [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+ * the second will correspond to ``Filter`` part of opertor
+ * the third will corresponde to ``Project`` part of operator
+
+Cost is also printed in ``EXPLAIN ANALYZE`` output.
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/optimizer/cost-in-explain.rst]: It is estimated for both of the inputs.

I changed the sentence to 
```
Estimated cost is also printed in ``EXPLAIN ANALYZE`` output as an addition to actual runtime statistics.
```

**Comment by losipiuk** [comment]
@akshatnair I addressed your comments.
@electrum please review.

**Comment by raghavsethi** [comment]
Can this still be reviewed in it's current form? Are there parts of this that are out of date?

**Comment by kokosing** [comment]
I think it can be reviewed.

**Pull request closed without merging**